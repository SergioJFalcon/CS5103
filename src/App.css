.App {
  align-items: center;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.header {
  width: 95%;
  text-align: right;
  margin: 10px;
}
.container {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
}

input[type="file"] {
  position: absolute;
  display: none;
  z-index: -1;
  top: 10px;
  left: 15px;
  font-size: 17px;
  color: #b8b8b8;

}

.button-wrap {
  position: relative;
}

.button {
  display: inline-block;
  padding: 12px 18px;
  cursor: pointer;
  border-radius: 5px;
  background-color: #8ebf42;
  font-size: 16px;
  font-weight: bold;
  color: #fff;
}
.button:hover {
  background-color: #00ab97;
}
.function_container {
  display: flex;
  justify-content: center;
}
.column {
  margin-left: 20px;
  margin-right: 20px;
}
.text {
  margin: 20px;
}
@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
