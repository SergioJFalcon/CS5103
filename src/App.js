/**
  Author: Sergio J Falcon
  Date: March 22nd, 2021
  Course: CS5103 - Software Engineering
  Project: Word Statistics
**/

import React, { useState, useEffect } from 'react';
import { useInput } from './hooks/input-hook';

import { printCleanList, cleanList, wordCount, wordFrequency, lineCount, charCount, uniqueCharFrequency, replaceWord } from './manipulation';

import './App.css';


function App() {
  const [text, setText] = useState('');
  const [fileName, setFileName] = useState('');

  const { value: previousWord, bind:bindPreviousWord, reset:resetPreviousWord} = useInput('');
  const { value: replacementWord, bind:bindReplacementWord, reset:resetReplacementWord} = useInput('');


  useEffect(() => {
    document.title = `File name is ${fileName}`;
  })

  const handleFiles = (e) => {
    const reader = new FileReader();
    let file = e.target.files[0];
    setFileName(file.name);
    
    reader.readAsText(e.target.files[0]);

    reader.onload = function () {
      setText(reader.result);
    };
    reader.onerror = function () {
      console.log(reader.error);
    };
  }

  const handleSubmit = event => {
    event.preventDefault();

    setText(replaceWord(text, previousWord, replacementWord));
    resetPreviousWord();
    resetReplacementWord();
  }
    return (
      <div className="App">
      <div className="header">
        <h4>Created by: Sergio J Falcon</h4>
      </div>
      {replaceWord('one one two three', 'one,', 'first')}

        {!fileName ? <h1>Upload a file to run Word Statistics</h1>
        : <h1>File uploaded: {fileName}</h1>}
        <div className="container">
          <div className="button-wrap">
            <label className="button" htmlFor="uploadFile">Upload File</label>
            <input id="uploadFile" type="file" onChange={(e) => handleFiles(e, 0)}></input>
          </div>
        </div>
        <div className='text'>
          {text ? printCleanList(cleanList(text)) : <p>empty</p>}
        </div>
        <div className="function_container">
          <div className="column">
            <h2>Word Count: </h2><p>{wordCount(text)}</p>
          </div>
          <div className="column">
            <h2>Word Frequency: </h2>
              {text ? (wordFrequency(text).map(entry =>(
                <p>{entry.word}: {entry.counter}</p>
              ))) : null}
          </div>
          <div className="column">
            <h2>Line Count: </h2>
            <p>{lineCount(text)}</p>
          </div>
          <div className="column">
            <h2>Char Count: </h2><p>{charCount(text)}</p>
          </div>
          <div className="column">
            <h2>Unique Char Frequency: </h2>
            {text ? (uniqueCharFrequency(text).map(entry => (
              <p>{entry.character}: {entry.counter}</p>
            ))) : null}
          </div>
          <div className="column">
              <h2>Word Replacement: </h2>
              <form onSubmit={handleSubmit}>
                <div>
                  <label>Word to be replaced: </label>
                  <input type="text" {...bindPreviousWord} />
                </div>
                
                <div>
                  <label>Replacement for said word:</label>
                  <input type="text" {...bindReplacementWord} />
                </div>
                <input type="submit" value="Submit" />
              </form>
              <p>{text}</p>
          </div>
        </div>
      </div>
    );
}

export default App;
