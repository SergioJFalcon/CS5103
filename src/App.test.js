import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';
import { create } from 'react-test-renderer'

describe('My first snapshot test', ()=> {
  test('testing app', () => {
    let tree = create(<App />);
    expect(tree.toJSON()).toMatchSnapshot();
  })
});
describe('Inputting a text file', () => {
  // test('input textfile', () => {
  //   // const component = render(<App />);

  //   // fireEvent.click(screen.getByText('Upload File'));
  // });
  test('handleFiles', () => {
    
  });
  test('handleSubmit', () => {

  });
  test('has correct body', () => {
    render(<App />);
    let headertwo = [<h2 title="word-count">Word Count: </h2>, <h2>Word Frequency: </h2>, <h2>Line Count: </h2>, <h2>Char Count: </h2>, <h2>Unique Char Frequency: </h2>, <h2>Word Replacement: </h2>]
    // let wordCountHeader = <h2 t)
    // expect(screen.getByRole('heading', { level: 4 })).toHaveTextContent('Created by: Sergio J Falcon');
    // expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('Upload a file to run Word Statistics');
    // expect(screen.queryAllByRole('heading', { level: 2 })).toHaveTextContent(headertwo);
    // console.log(getByText('Word Count: ', 'h2'))
    //expect(screen.getAllByTitle('word-count')).toHaveTextContent('Word Count: ');

  })
});