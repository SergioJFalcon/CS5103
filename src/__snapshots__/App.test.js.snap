// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first snapshot test testing app 1`] = `
<div
  className="App"
>
  <div
    className="header"
  >
    <h4>
      Created by: Sergio J Falcon
    </h4>
  </div>
  first first two three
  <h1>
    Upload a file to run Word Statistics
  </h1>
  <div
    className="container"
  >
    <div
      className="button-wrap"
    >
      <label
        className="button"
        htmlFor="uploadFile"
      >
        Upload File
      </label>
      <input
        id="uploadFile"
        onChange={[Function]}
        type="file"
      />
    </div>
  </div>
  <div
    className="text"
  >
    <p>
      empty
    </p>
  </div>
  <div
    className="function_container"
  >
    <div
      className="column"
    >
      <h2
        title="word-count"
      >
        Word Count: 
      </h2>
      <p />
    </div>
    <div
      className="column"
    >
      <h2>
        Word Frequency: 
      </h2>
    </div>
    <div
      className="column"
    >
      <h2>
        Line Count: 
      </h2>
      <p />
    </div>
    <div
      className="column"
    >
      <h2>
        Char Count: 
      </h2>
      <p />
    </div>
    <div
      className="column"
    >
      <h2>
        Unique Char Frequency: 
      </h2>
    </div>
    <div
      className="column"
    >
      <h2>
        Word Replacement: 
      </h2>
      <form
        onSubmit={[Function]}
      >
        <div>
          <label>
            Word to be replaced: 
          </label>
          <input
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <div>
          <label>
            Replacement for said word:
          </label>
          <input
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
        <input
          type="submit"
          value="Submit"
        />
      </form>
      <p>
        
      </p>
    </div>
  </div>
</div>
`;
